src/api/modelTypes.js

```diff
@@ -350,19 +350,16 @@ export type Reaction = $ReadOnly<{|
 export type MessageSnapshot = $ReadOnly<{|
   user_id: number,
   timestamp: number,
+  topic: string,
+  content: string,
+  rendered_content: string,
 
-  /** Docs unclear but suggest absent if only content edited. */
-  topic?: string,
-
-  /**
-   * Docs unclear, but suggest these five absent if only topic edited.
-   * They definitely say "prev"/"diff" properties absent on the first snapshot.
-   */
-  content?: string,
-  rendered_content?: string,
+  // The prev_* field is only present if the respective attribute
+  // of the message was changed.
   prev_content?: string,
   prev_rendered_content?: string,
   content_html_diff?: string,
+  prev_topic?: string,
```

Comment:
This summary is concise in a similar way to the corresponding bit of the API docs -- but I think it would actually be clearer (and not even many more words, if more at all) to be less abstract, more concrete and explicit:
```js
  // These are present just if the content was edited.
  prev_content?: string,
￼  prev_rendered_content?: string,
￼  content_html_diff?: string,

  // Present just if the topic was edited.
  prev_topic?: string,
```

(... OK, and I just counted: that's 16 words instead of 15. :grin: )